core/src/main/java/io/atomix/core/Atomix.java

```diff
@@ -472,23 +471,38 @@ private static ManagedMessagingService buildMessagingService(AtomixConfig config
   private static ManagedBroadcastService buildBroadcastService(AtomixConfig config) {
     return NettyBroadcastService.builder()
         .withLocalAddress(config.getClusterConfig().getLocalNode().getAddress())
-        .withGroupAddress(config.getClusterConfig().getBootstrapConfig().getMulticastAddress())
-        .withEnabled(config.getClusterConfig().getBootstrapConfig().isMulticastEnabled())
+        .withGroupAddress(config.getClusterConfig().getMulticastAddress())
+        .withEnabled(config.getClusterConfig().isMulticastEnabled())
         .build();
   }
 
   /**
    * Builds a bootstrap metadata service.
    */
   private static ManagedBootstrapMetadataService buildBootstrapMetadataService(AtomixConfig config) {
-    return new DefaultBootstrapMetadataService(config.getClusterConfig().getBootstrapConfig());
+    boolean hasCoreNodes = config.getClusterConfig().getNodes().stream().anyMatch(node -> node.getType() == Node.Type.CORE);
+    ClusterMetadata metadata = ClusterMetadata.builder()
+        .withNodes(config.getClusterConfig().getNodes()
+            .stream()
+            .filter(node -> !hasCoreNodes || node.getType() == Node.Type.CORE)
```

Comment:
Do we really want clients in here too if no core nodes are defined?

Reply:
Hmm... probably not

Reply:
the way I read it at least is that if no core nodes are defined client nodes will be considered bootstrap nodes.

Reply:
That is indeed true

Reply:
Fixed